fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -3928,6 +3928,12 @@ private void truncateTableInternal(OlapTable olapTable, List<Partition> newParti
         for (Long tabletId : oldTabletIds) {
             GlobalStateMgr.getCurrentInvertedIndex().deleteTablet(tabletId);
         }
+
+        // if it is lake table, need to delete shard and drop tablet
+        if (olapTable.isLakeTable()) { 
+            stateMgr.getShardManager().getShardDeleter().addUnusedShardId(oldTabletIds);
+            editLog.logAddUnusedShard(oldTabletIds);
+        }
```

Comment:
Put this outside `truncateTableInternal` because `replayTruncateTable` also calls this method.